bus/api.go

```diff
@@ -146,21 +146,30 @@ type AddObjectRequest struct {
 // DownloadParams contains the metadata needed by a worker to download an object.
 type DownloadParams struct {
 	ContractSet string
+	GougingParams
 }
 
 // UploadParams contains the metadata needed by a worker to upload an object.
 type UploadParams struct {
 	CurrentHeight uint64
-	MinShards     uint8
-	TotalShards   uint8
 	ContractSet   string
+	GougingParams
 }
 
 // MigrateParams contains the metadata needed by a worker to migrate a slab.
 type MigrateParams struct {
 	CurrentHeight uint64
 	FromContracts string
 	ToContracts   string
+	GougingParams
+}
+
+// GougingParams contains the metadata needed by a worker to perform gouging
+// checks.
+type GougingParams struct {
+	Period             uint64
```

Comment:
The easiest way to get rid of this is probably through splitting up `MaxUploadPrice` and `MaxStoragePrice`. Then we make `MaxStoragePrice` the max price per TiB per block. If we break it down to a price-per-block we don't need the period anymore.

Reply:
Good idea!